railties/lib/rails/generators.rb

```diff
@@ -274,8 +274,9 @@ def invoke(namespace, args = ARGV, config = {})
         else
           options     = sorted_groups.flat_map(&:last)
           suggestions = options.sort_by { |suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
+          suggestions.map! { |s| "'#{s}'" }
           msg =  "Could not find generator '#{namespace}'. ".dup
-          msg << "Maybe you meant #{ suggestions.map { |s| "'#{s}'" }.to_sentence(last_word_connector: " or ", locale: :en) }\n"
+          msg << "Maybe you meant #{ suggestions[0...-1].join(', ')} or #{suggestions[-1]}\n"
```

Comment:
Is not this add a trailing `or` when there is only one suggestion?

Reply:
The current suggestion implementation sorts the options using levenshtein distance and uses the first three of the results. 
https://github.com/rails/rails/blob/89a209f1abba5a2320d31c4898dea150c0abd0c0/railties/lib/rails/generators.rb#L276 

This is just sorting, the number of options itself does not change. Therefore, even if it is a mistaken value as a suggestion, three are always suggested. 

``` 
$ ./bin/rails g z
Could not find generator 'z'. Maybe you meant 'job', 'task' or 'model'
Run `rails generate --help` for more options. 
```